frontend/src/pages/SignupPage/index.js

```diff
@@ -16,14 +16,64 @@ const STEP_3 = "STEP_3";
 const SignupPage = () => {
   const navigator = useNavigate();
 
-  const [step, setStep] = useState(STEP_1);
-  const [name, setName] = useState("");
-  const [email, setEmail] = useState("");
-  const [selectedInterests, setSelectedInterests] = useState([]);
-  const [addedInterest, setAddedInterest] = useState("");
-  const [error, setError] = useState({ hasError: false, message: "" });
+  const [state, setState] = useState({
+    step: STEP_1,
+    name: "",
+    email: "",
+    selectedInterests: [],
+    addedInterest: "",
+    error: { hasError: false, message: "" },
+  });
 
-  const handleStep = () => {
+  const setError = ({ hasError, message }) => {
+    setState((prevState) => ({
+      ...prevState,
+      error: { hasError: hasError, message: message },
+    }));
+  };
+
+  const setSelectedInterests = (id) => {
+    if (state.selectedInterests.includes(id)) {
+      setState((prevState) => ({
```

Comment:
Why are you using `prevState` everywhere?

Reply:
When I used things like setState({ name: newName }) it deleted my object structure and made it as only made up by a name. So I'd lose all of the other attributes. The way I saw that I could change a single attribute without losing the others was using prevState